src/libostree/ostree-sepolicy.c

```diff
@@ -599,6 +599,25 @@ ostree_sepolicy_get_label (OstreeSePolicy    *self,
   return TRUE;
 }
 
+// If policy doesn't specify a label, try a fallback.
+gboolean
+_ostree_sepolicy_require_label (OstreeSePolicy *policy, const char *relpath,
+                                guint32 unix_mode, char **out_label, 
+                                GCancellable *cancellable, GError **error)
+{
+  char *label = NULL;
+  if (!ostree_sepolicy_get_label (policy, relpath, unix_mode, &label, cancellable, error))
+    return FALSE;
+  if (!label)
+    {
+      if (!ostree_sepolicy_get_label (policy, "/usr/share/some-generic-thing", unix_mode, &label, cancellable, error))
```

Comment:
I think a more appropriate thing to do here would be to keep going up from `relpath` and stopping until we have a parent dir which has a label. This more closely matches the default heuristic used by SELinux.

Reply:
I'm not sure.  In the case of this `/var/tmp` file, yes.   On the other hand, if someone is trying to add `/foobar`, I don't think we want it to be `root_t`.

Reply:
I think it should be safe to trust the policy. E.g. `/foobar` would get caught by https://github.com/fedora-selinux/selinux-policy/blob/1715509773b3387a0c74423c05d53eb401d9b470/policy/modules/kernel/files.fc#L5.